src/command/dev/protocol/leader.rs

```diff
@@ -283,11 +283,13 @@ impl LeaderSession {
     fn compose(&mut self) -> CompositionResult {
         self.compose_runner
             .run(&mut self.supergraph_config())
-            .map(|maybe_new_schema| {
+            .and_then(|maybe_new_schema| {
                 if maybe_new_schema.is_some() {
-                    let _ = self.router_runner.spawn().map_err(|e| panic!("{}", e));
+                    if let Err(err) = self.router_runner.spawn() {
```

Comment:
nice. might be worth bumping the ready check timeout to be longer than 5 seconds too, what do you think

Reply:
Hmm, there's definitely a tradeoff with having to wait longer to see problems. I think _most_ routers should be ready within 5 seconds, maybe we add a flag to make that configurable for more complex setups?

Reply:
a flag seems reasonable - i think 5 seconds should be long enough as well, hopefully just reporting router output here helps track down the issue

Reply:
i'll merge as is and if we see timeout issues rolling in instead of panics we can take a look